src/DataStructures/DataBox/Prefixes.hpp

```diff
@@ -14,32 +16,123 @@
+ * parameters:
+ * 1. The tag to wrap
+ * 2. The volume dim as a type (e.g. `tmpl::size_t<Dim>`)
+ * 3. The frame of the derivative index in `Tensor`s
+ *
+ * The partial derivatives in a Frame that is not the logical frame as a compute
+ * item is requested by specifying the list of tags in the
+ * `Tags::Variables<tmpl::list<Tags...>>`, the list of tags to be
+ * differentiated, and the Tag for inverse Jacobian between the logical frame
+ * and the frame in which the derivative is taken. Note that the derivative Tags
+ * must be a contiguous subset from the head of the `Tags...` pack.
+ *
+ * The partial derivatives in the logical frame can also be computed. In that
+ * case the template parameters must be the list of Variables tags and
+ * derivative tags, with the last parameter being a `std::integral_constant` of
+ * the dimension of the mesh.
+ */
+template <typename Tag, typename VolumeDim, typename Frame, typename>
+struct deriv;
+
+template <typename Tag, typename VolumeDim, typename Frame>
+struct deriv<Tag, VolumeDim, Frame,
+             Requires<tt::is_a_v<Tensor, db::item_type<Tag>>>>
     : db::DataBoxPrefix {
   using type = TensorMetafunctions::prepend_spatial_index<
-      db::item_type<Tag>, VolumeDim::value, UpLo::Lo, Fr>;
+      db::item_type<Tag>, VolumeDim::value, UpLo::Lo, Frame>;
   using tag = Tag;
   static constexpr db::DataBoxString label = "d";
 };
 
-template <typename Tag, typename VolumeDim, typename Fr>
-struct d<Tag, VolumeDim, Fr,
-         Requires<tt::is_a_v<::Variables, db::item_type<Tag>>>>
+template <typename Tag, typename VolumeDim, typename Frame>
+struct deriv<Tag, VolumeDim, Frame,
+             Requires<tt::is_a_v<::Variables, db::item_type<Tag>>>>
     : db::DataBoxPrefix {
   using type = db::item_type<Tag>;
   using tag = Tag;
   static constexpr db::DataBoxString label = "d";
 };
 
+// Partial derivatives with derivative index in frame related to by
+// InverseJacobianTag
+template <typename... VariablesTags, typename... DerivativeTags,
+          typename InverseJacobianTag>
+struct deriv<tmpl::list<VariablesTags...>, tmpl::list<DerivativeTags...>,
+             InverseJacobianTag,
+             Requires<tt::is_a_v<Tensor, db::item_type<InverseJacobianTag>>>>
+    : db::ComputeItemTag {
+ private:
+  using derivative_frame_index =
+      tmpl::back<typename db::item_type<InverseJacobianTag>::index_list>;
+  using variables_tags = tmpl::list<VariablesTags...>;
+
+ public:
+  static constexpr db::DataBoxString label = "d";
+  static constexpr auto function =
+      partial_derivatives<tmpl::list<DerivativeTags...>, variables_tags,
+                          derivative_frame_index::dim,
+                          typename derivative_frame_index::Frame>;
+  using argument_tags =
+      typelist<Tags::Variables<variables_tags>,
+               Tags::Extents<derivative_frame_index::dim>, InverseJacobianTag>;
+};
+
+// Logical partial derivatives
```

Comment:
Do we have a known use for derivatives in the logical coordinates?  I expected them to be used to calculate derivatives in other frames, but it looks like we're not doing that.